File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        if (nums.length < 2) return nums.length;

        int max = 1;
        for (int i : nums) {
            set.add(i);
        }
        for (int i : nums) {
            if (!set.contains(i - 1)) {
                int j = 1;
                while (set.contains(i + j)) {
                    j++;
                }
                max = Math.max(max, j);
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1
        if nums[r] > nums[l] or len(nums) == 1:
            return nums[l]
        while l < r:
            m = l + (r - l) // 2
            if nums[m] > nums[l]:
                l = m
            else:
                r = m
        return nums[r + 1]
        